src/blockdata/opcodes.rs

```diff
@@ -674,77 +674,62 @@ impl All {
-                // 1 opcode
-                } else if self == all::OP_PUSHNUM_NEG1 {
-                    Class::PushNum(-1)
-                // 16 opcodes
-                } else if all::OP_PUSHNUM_1.code <= self.code &&
-                        self.code <= all::OP_PUSHNUM_16.code {
-                    Class::PushNum(1 + self.code as i32 - all::OP_PUSHNUM_1.code as i32)
-                // 76 opcodes
-                } else if self.code <= all::OP_PUSHBYTES_75.code {
-                    Class::PushBytes(self.code as u32)
-                // 61 opcodes
-                } else {
-                    Class::Ordinary(Ordinary::try_from_all(self).unwrap())
-                },
-            ClassifyContext::Legacy =>
-                if self == all::OP_VERIF || self == all::OP_VERNOTIF ||
-                    self == all::OP_CAT || self == all::OP_SUBSTR ||
-                    self == all::OP_LEFT || self == all::OP_RIGHT ||
-                    self == all::OP_INVERT || self == all::OP_AND ||
-                    self == all::OP_OR || self == all::OP_XOR ||
-                    self == all::OP_2MUL || self == all::OP_2DIV ||
-                    self == all::OP_MUL || self == all::OP_DIV || self == all::OP_MOD ||
-                    self == all::OP_LSHIFT || self == all::OP_RSHIFT {
-                    Class::IllegalOp
-                // 11 opcodes
-                } else if self == all::OP_NOP ||
-                        (all::OP_NOP1.code <= self.code &&
-                            self.code <= all::OP_NOP10.code) {
-                    Class::NoOp
-                // 75 opcodes
-                } else if self == all::OP_RESERVED || self == all::OP_VER || self == all::OP_RETURN ||
-                        self == all::OP_RESERVED1 || self == all::OP_RESERVED2 ||
-                        self.code >= all::OP_CHECKSIGADD.code {
-                    Class::ReturnOp
-                // 1 opcode
-                } else if self == all::OP_PUSHNUM_NEG1 {
-                    Class::PushNum(-1)
-                // 16 opcodes
-                } else if all::OP_PUSHNUM_1.code <= self.code &&
-                        self.code <= all::OP_PUSHNUM_16.code {
-                    Class::PushNum(1 + self.code as i32 - all::OP_PUSHNUM_1.code as i32)
-                // 76 opcodes
-                } else if self.code <= all::OP_PUSHBYTES_75.code {
-                    Class::PushBytes(self.code as u32)
-                // 61 opcodes
-                } else {
-                    Class::Ordinary(Ordinary::try_from_all(self).unwrap())
-                }
+        use self::all::*;
+        match (self, ctx) {
+            // 3 opcodes illegal in all contexts
+            (OP_VERIF, _) | (OP_VERNOTIF, _) | (OP_INVALIDOPCODE, _) => Class::IllegalOp,
+
+            // 15 opcodes illegal in Legacy context
+            (OP_CAT, ctx) | (OP_SUBSTR, ctx) 
+            | (OP_LEFT, ctx) | (OP_RIGHT, ctx) 
+            | (OP_INVERT, ctx) 
```

Comment:
Trailing whitespace for the above three lines

Reply:
Fixed as a high priority